Source/Core/DolphinQt/GBAWidget.cpp

```diff
@@ -166,6 +166,23 @@ void GBAWidget::UnloadROM()
   RestartCore(m_core);
 }
 
+void GBAWidget::QueueEReaderCard()
+{
+  QString card_path = QDir::toNativeSeparators(QFileDialog::getOpenFileName(
+      this, tr("Select a File"), QString(), tr("e-Reader Cards (*.raw);;All Files (*)"), nullptr,
+      QFileDialog::Options()));
+
+  if (card_path.isEmpty())
+    return;
+
+  Core::RunOnCPUThread(
+      [core = m_core, card_path = card_path.toStdString()] {
+        if (auto core_ptr = core.lock())
+          core_ptr->EReaderQueueCard(card_path);
+      },
+      false);
```

Comment:
Since this shows up more than once now it might be better to move into its own function